scene/register_scene_types.cpp

```diff
@@ -170,9 +173,6 @@
 #include "scene/resources/world_2d.h"
 #include "scene/scene_string_names.h"
 
-#include "scene/3d/spatial.h"
-#include "scene/3d/world_environment.h"
```

Comment:
Those were outside the `#ifndef` because they're also needed when 3D is disabled. Did you try it out to confirm if that changed?

You didn't move their `register_class` calls though so I think it might still be working if those headers are included anyway through other headers, but I'm pretty sure this move is not correct.

Reply:
Just reverted it to be safe. This is dangerous territory, I'll trust your judgement. It didn't seem to break anything though.